toolchain/check/testdata/operators/overloaded/index.carbon

```diff
@@ -24,7 +24,7 @@ library "[[@TEST_NAME]]";
 
 import Core library "core_wrong_index_with";
 
-// CHECK:STDERR: fail_wrong_index_with.carbon:[[@LINE+4]]:10: error: type `Core.IntLiteral` does not support indexing [TypeNotIndexable]
+// CHECK:STDERR: fail_wrong_index_with.carbon:[[@LINE+4]]:10: error: value of type `type` is not callable [CallToNonCallable]
```

Comment:
This error is a bit less helpful, I wonder if we could improve things in this case?